### DISCUSSION TITLE
WARNING  Overriding detected file kind

### DISCUSSION THREAD

**Discussion by feuertaenzer** [ansible/ansible-lint#1679] [Q&A]
Hey folks,
We recently noticed a bunch of warnings when running lint, regarding our playbooks:
`WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: some.yml`

We get this warning for every playbook in the top level directory(if that's the correct term).
I understand that the some.yml is initially detected as just a random yml-file and then correctly redetected as a playbook but 
that step wasn't necessary before. The yml-files are located in the top level directory as usual(as far as I know that's the usual location). Roles and everything else are located in subfolders and are properly detected without warnings.

So far it doesn't cause any issues but still we'd like to fix it. For some reason it's super hard to find information on this. I found it mentioned in a few issues but it was pretty much always just ignored.

Does anyone know or have a guess what we have to do?

**Comment by ozaryx** [comment]
HI,

This came from this code

https://github.com/ansible-community/ansible-lint/blob/main/src/ansiblelint/config.py#L13-L33

**Reply by Grummel7** [reply]
Looking at the code, it seems I need to put my playbooks into a directory called playbooks to get rid of the warning:
```
$ ansible-lint products/ccms/playbooks/get_status.yml 
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: products/ccms/playbooks/get_status.yml
```
Well, playbooks in this example was symlink, since I do not want to change my project structure to satisfy the whim of a test tool.

But indeed when I rename ccms to playbooks, the warning is gone.

**Comment by thenger** [comment]
to remove warning you can also specify that playbooks are ending with "yaml" in the ~/.ansible-lint file

#### ~/.ansible-lint
``` yaml
# vim: set ft=yaml:
---
kinds:
  - playbook: "**/*.{yml,yaml}"
...
```
source: [ansible-lint.readthedocs.io/configuring](https://ansible-lint.readthedocs.io/configuring/)
hth

**Reply by fetzerch** [reply]
This setting causes all `*.yml/yaml` files (including roles, hosts, ...) to be treated as playbooks. And because `kinds` matches against an [absolute path](https://github.com/ansible/ansible-lint/blob/497bae917df7739eae885b219440e4be0ee31bcb/src/ansiblelint/file_utils.py#L143), I currently don't see an easy option to just match the top level `*.yml/yaml` files.
